src/overlays/fbdemos/ovl_fbdemo_wipe3/z_fbdemo_wipe3.c

```diff
@@ -17,37 +19,141 @@ void TransitionWipe3_SetColor(void* thisx, u32 color);
 void TransitionWipe3_SetEnvColor(void* thisx, u32 color);
 s32 TransitionWipe3_IsDone(void* thisx);
 
-#if 0
+#include "assets/overlays/ovl_fbdemo_wipe3/ovl_fbdemo_wipe3.c"
+
+TexturePtr sTransWipe3Textures[] = {
+    fbdemo_tex_000520, fbdemo_tex_000920, fbdemo_tex_000D20, fbdemo_tex_001120,
+    fbdemo_tex_001520, fbdemo_tex_000520, fbdemo_tex_000520, fbdemo_tex_000520,
+};
+
 TransitionInit TransitionWipe3_InitVars = {
-    TransitionWipe3_Init,
-    TransitionWipe3_Destroy,
-    TransitionWipe3_Update,
-    TransitionWipe3_Draw,
-    TransitionWipe3_Start,
-    TransitionWipe3_SetType,
-    TransitionWipe3_SetColor,
-    TransitionWipe3_SetEnvColor,
-    TransitionWipe3_IsDone,
+    TransitionWipe3_Init,     TransitionWipe3_Destroy,     TransitionWipe3_Update,
+    TransitionWipe3_Draw,     TransitionWipe3_Start,       TransitionWipe3_SetType,
+    TransitionWipe3_SetColor, TransitionWipe3_SetEnvColor, TransitionWipe3_IsDone,
 };
 
-#endif
+void TransitionWipe3_Start(void* thisx) {
+    TransitionWipe3* this = THIS;
+
+    this->isDone = false;
+    this->curTexture = sTransWipe3Textures[this->texIndex & 7];
+    if (this->unk_16 == 0) {
+        this->wipeSpeed = 20;
+    } else {
+        this->wipeSpeed = 10;
+    }
+    if (this->unk_15 == 0) {
+        this->color.rgba = RGBA8(0, 0, 0, 255);
+    } else if (this->unk_15 == 1) {
+        this->color.rgba = RGBA8(255, 255, 255, 255);
+    } else {
+        this->color.rgba = RGBA8(160, 160, 160, 255);
+    }
+    if (this->unk_14) {
+        this->scrollY = 0;
+    } else {
+        this->scrollY = 500;
+        if (this->texIndex == 2) {
+            func_8019F128(NA_SE_OC_SECRET_WARP_OUT);
+        }
+    }
+    guPerspective(&this->projection, &this->normal, 60.0f, 1.3333334f, 10.0f, 12800.0f, 1.0f);
+    guLookAt(&this->lookAt, 0.0f, 0.0f, 400.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 0.0f);
+}
+
+void* TransitionWipe3_Init(void* thisx) {
+    TransitionWipe3* this = THIS;
+
+    bzero(this, sizeof(TransitionWipe3));
+    return this;
+}
+
+void TransitionWipe3_Destroy(void* thisx) {
+}
+
+void TransitionWipe3_UpdateScrollY(void* thisx, f32 scroll) {
+    TransitionWipe3* this = THIS;
+
+    this->scrollY = (s32)(500.0f * scroll);
+}
+
+void TransitionWipe3_Update(void* thisx, s32 updateRate) {
+    TransitionWipe3* this = THIS;
+
+    if (this->unk_14) {
+        if ((this->scrollY == 0) && (this->texIndex == 2)) {
+            func_8019F128(NA_SE_OC_SECRET_WARP_IN);
+        }
+        this->scrollY += (this->wipeSpeed * 3) / updateRate;
+        if (this->scrollY >= 500) {
+            this->scrollY = 500;
+            this->isDone = true;
+        }
+    } else {
+        this->scrollY -= (this->wipeSpeed * 3) / updateRate;
+        if (this->scrollY <= 0) {
+            this->scrollY = 0;
+            this->isDone = true;
+        }
+    }
+}
+
+void TransitionWipe3_Draw(void* thisx, Gfx** gfxP) {
+    Gfx* dl = *gfxP;
```

Comment:
```suggestion
    Gfx* gfx = *gfxP;
```
Generally we have been using just `gfx` for these.